### PR TITLE
Added Algolia search

### PR DESCRIPTION
**Pull request by jroper**
Added Algolia search. Also ensured that when anchor links are navigated to, the sticky header doesn't cover the top of the anchor.

### CODE REVIEW

File: src/main/assets/_documentation.styl [hunk 1]
```diff
@@ -1,4 +1,12 @@
 article
+
+  /* This takes into account the header when clicking anchor links */
+  :target:before
+    content ""
+    display block
+    height header-height
+    margin-top - header-height
+
```
Comment (jroper) [inline src/main/assets/_documentation.styl]: @rasummer Is this ok?  Basically, when you navigated to an anchor in the docs, the title would be chopped off by the sticky header, so I've added this to fix it, which seems to work to me, but you probably have a better idea of the best way to address it.

Comment (rasummer) [inline src/main/assets/_documentation.styl]: @jroper This works nicely.

Comment (jroper) [inline src/main/assets/_documentation.styl]: @rasummer It seems this doesn't work on Edge - would you recommend anything else?

Comment (jroper) [inline src/main/assets/_documentation.styl]: Actually, it does, but I've found some problems in our own code.

**Pull request merged**